app/javascript/shared/components/tags.jsx

```diff
@@ -1,502 +1,135 @@
-      !this.isBottomOfSearchResults &&
-      searchResults.length > 0
-    ) {
-      e.preventDefault();
-      this.moveDownInSearchResults();
-    } else if (e.key === KEYS.UP && !this.isTopOfSearchResults) {
-      e.preventDefault();
-      this.moveUpInSearchResults();
-    } else if (e.key === KEYS.RETURN && this.isSearchResultSelected) {
-      e.preventDefault();
-      this.insertTag(
-        component.state.searchResults[component.state.selectedIndex].name,
-      );
-
-      setTimeout(() => {
-        document.getElementById('tag-input').focus();
-      }, 10);
-    } else if (e.key === KEYS.COMMA && !this.isSearchResultSelected) {
-      this.fetchTopTagSuggestions();
-      this.clearSelectedSearchResult();
-    } else if (e.key === KEYS.DELETE) {
-      if (
-        component.props.defaultValue[
-          component.props.defaultValue.length - 1
-        ] === ','
-      ) {
-        this.clearSelectedSearchResult();
-      }
-    } else if (
-      !LETTERS_NUMBERS.test(e.key) &&
-      !NAVIGATION_KEYS.includes(e.key)
-    ) {
-      e.preventDefault();
-    }
-  };
-
-  handleRulesClick = (e) => {
-    e.preventDefault();
-    const { showingRulesForTag } = this.state;
-    if (showingRulesForTag === e.target.dataset.content) {
-      this.setState({ showingRulesForTag: null });
-    } else {
-      this.setState({ showingRulesForTag: e.target.dataset.content });
-    }
-  };
-
-  handleTagClick = (e) => {
-    if (e.target.className === 'articleform__tagsoptionrulesbutton') {
-      return;
-    }
-
-    const input = document.getElementById('tag-input');
-    input.focus();
-
-    // the rules container (__tagoptionrow) is the real target of the event,
-    // by using currentTarget we let the event propagation work
-    // from the inner rules box as well (__tagrules)
-    this.insertTag(e.currentTarget.dataset.content);
-  };
-
-  handleInput = (e) => {
-    let { value } = e.target;
-    // If we start typing immediately after a comma, add a space
-    // before what we typed.
-    // e.g. If value = "javascript," and we type a "p",
-    // the result should be "javascript, p".
-    if (e.inputType === 'insertText') {
-      const isTypingAfterComma =
-        e.target.value[e.target.selectionStart - 2] === ',';
-      if (isTypingAfterComma) {
-        value = this.insertSpace(value, e.target.selectionStart - 1);
-      }
-    }
-
-    if (e.data === ',') {
-      value += ' ';
+import { useTagsField } from '../../hooks/useTagsField';
+import { TagAutocompleteOption } from '../../article-form/components/TagAutocompleteOption';
+import { TagAutocompleteSelection } from '../../article-form/components/TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+
+export const Tags = ({
+  defaultValue,
+  listing,
+  onInput,
+  classPrefix,
+  categorySlug,
+  maxTags,
+  name,
+  onFocus,
+}) => {
+  const listingState = useMemo(
+    () =>
+      listing
```

Comment:
We can remove a lot of old conditionals from this code - since this file is _only_ being used for listings now, we don't need to pass it in as a prop or check for it here

Reply:
I agree!